assets/app/scripts/services/applicationGenerator.js

```diff
@@ -202,6 +202,13 @@ angular.module("openshiftConsole")
           type: "ImageChange"
         });
       }
+
+      // User can input a URL that contains a ref
+      var urlAndRef = input.buildConfig.sourceUrl.split("#");
```

Comment:
I thought of suggesting this, but we accept so many different forms of URIs. I was afraid one might not parse. For instance, does `git@example.com:dir1/dir2` parse in the browser as a URI?